ttl.py

```diff
@@ -0,0 +1,153 @@
+"""Module for TTL status management."""
+
+import asyncio
+import dataclasses
+import enum
+import time
+from typing import Any
+
+from artiq.coredevice.comm_moninj import CommMonInj, TTLOverride, TTLProbe
+
+from protocols import SortedQueue
+
+device_to_channel: dict[str, int] = {}
+channel_to_device: dict[int, str] = {}
+
+def map_device_channel(ttl_devices: list[str], device_db: dict[str, Any]):
+    """Maps TTL devices and channels.
+    
+    Args:
+        ttl_devices: See main.configs.
+        device_db: See main.device_db.
+    """
+    for device in ttl_devices:
+        channel = device_db[device]["arguments"]["channel"]
+        device_to_channel[device] = channel
+        channel_to_device[channel] = device
```

Comment:
I think this can be packed within a class, e.g., `DeviceChannelMapping`, which provides methods `device(channel: int)` and `channel(device: str)`.
Then `map_device_channel()` would be the constructor of the class.
This design has a benefit that many mappings can exist and it does not use global (module-level) objects.